### PR TITLE
Add basic tests for passphrase retriever to make sure it handles delegations

### PR DESCRIPTION
**Pull request by cyli**
Not a ton of coverage, but just wanted to make sure it would work when we created delegation keys, signed stuff with them, etc.

### CODE REVIEW

File: passphrase/passphrase_test.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	expectedText := []string{
+		`Enter passphrase for new targets/a key with ID 0123456 (repo): `,
+		`Repeat passphrase for new targets/a key with ID 0123456 (repo):`,
+	}
+
+	assert.Equal(t, expectedText, lines)
+}
+
+// PromptRetrieverWithInOut, if asked for root, targets, delegation, and
+// snapshot keys in that order will only prompt for root, targets, and snapshot
```
Comment (riyazdf) [inline passphrase/passphrase_test.go]: should the comment read `...only prompt for root, targets, and delegation`?  (replacing snapshot with delegation)

Comment (cyli) [inline passphrase/passphrase_test.go]: Yes, thanks for that catch!

**Comment by riyazdf** [comment]
Awesome tests, LGTM!

File: passphrase/passphrase.go [hunk 2]
```diff
@@ -99,12 +99,14 @@ func PromptRetrieverWithInOut(in io.Reader, out io.Writer, aliasMap map[string]s
 			return "", true, ErrTooManyAttempts
 		}
 
-		state, err := term.SaveState(0)
-		if err != nil {
-			return "", false, err
+		if term.IsTerminal(0) {
```
Comment (diogomonica) [inline passphrase/passphrase.go]: comment appreciated on what we're trying to acomplish

**Comment by diogomonica** [comment]
LGTM

**Pull request merged**